src/BeamAdapter/component/mapping/AdaptiveBeamMapping.inl

```diff
@@ -626,28 +661,19 @@ void AdaptiveBeamMapping< TIn, TOut>::computeDistribution()
 
                     computeIdxAndBaryCoordsForAbs(b, xBary,  xAbs );
 
-                    PosPointDefinition beamDistrib;
-                    beamDistrib.beamId = b;
-                    beamDistrib.baryPoint[0] = xBary;
-                    beamDistrib.baryPoint[1] = points[i][1];
-                    beamDistrib.baryPoint[2] = points[i][2];
-
-                    m_pointBeamDistribution.push_back(beamDistrib);
+                    m_pointBeamDistribution.emplace_back( b, Vec3{ xBary, points[i][1], points[i][2]});
                 }
             }
         }
         else
         {
+            m_pointBeamDistribution.reserve(points.size());
+            const bool printLog = this->f_printLog.getValue();
             for (unsigned int i=0; i<points.size(); i++)
             {
-                PosPointDefinition beamDistrib;
-                beamDistrib.beamId = (int) floor(points[i][0]);
-                if ( (beamDistrib.beamId>numBeams-1 || beamDistrib.beamId<0.0 ) && this->f_printLog.getValue()  )
-                    msg_warning() <<"Points["<<i<<"][0] = "<<beamDistrib.baryPoint[0]<<" is defined outside of the beam length";
-                beamDistrib.baryPoint[0] = points[i][0] - floor(points[i][0]);
-                beamDistrib.baryPoint[1] = points[i][1];
-                beamDistrib.baryPoint[2] = points[i][2];
-                m_pointBeamDistribution.push_back(beamDistrib);
+                const unsigned int beamId = (int)floor(points[i][0]);
+                msg_warning_when(printLog && (beamId > numBeams-1)) << "Points[" << i << "][0] = " << beamId << " is defined outside of the beam length";
```

Comment:
I am not sure to understand why you want printLog to be true for a warning

Reply:
I wanted to keep the same condition as before, but I can remove it

Reply:
yes remove, a warning, is a warning ^^